src/npc.cpp

```diff
@@ -2955,7 +2960,7 @@ void npc::die( Creature *nkiller )
         // *only* set to true in this function!
         return;
     }
-
+    bool badguy = ( guaranteed_hostile() || !hit_by_player );
```

Comment:
This assignment doesn't make much sense. NPCs are constructed with `hit_by_player = false`, so this always evaluates true for one-hit kills. (e.g. Character::melee_attack_abstract-->Character::on_hit--->npc::die executes BEFORE npc::on_attacked)